### DISCUSSION TITLE
Having multiple Popovers/Modals/Panels, click-outside closing all

### DISCUSSION THREAD

**Discussion by jgodi** [floating-ui/floating-ui#2138] [React DOM] [answered]
Hello, we use FloatingUI in our component library for Tooltips/Popovers/Modals/Panels/Sidebars. We have some use cases where there is a Popover > Modal or Panel > Popover > Modal, and when you have a click outside then it closes all of them.

Is there a way to have each different floating element have its own closing lifecycle?

https://codesandbox.io/s/wizardly-dan-l60luz?file=/src/App.tsx is a good example of two popovers. Using the triggers, it works fine -- when you click outside then it closes both.

**Comment by atomiks** [comment, marked as answer]
Dismissals like `outsidePress` can change how the dismissal `bubbles` through a [`FloatingTree`](https://floating-ui.com/docs/FloatingTree): https://floating-ui.com/docs/useDismiss#bubbles

**Reply by jgodi** [reply]
Thanks!

**Reply by jgodi** [reply]
Quick follow up, I have [this](https://codesandbox.io/s/great-moon-ktuin0?file=/src/Popover.tsx:1322-1378) example updated to use `const dismiss = useDismiss(context, { bubbles: false });` but it still closes both Popovers.

The bubbles property mentions that there also needs to have focus inside to work, is there a way to get this to work without that? Our use cases wouldn't require focus to be within the popover, itll likely remain on the trigger.

**Reply by atomiks** [reply]
That's a limitation of it currently, but it's possible the implementation can be re-worked to not need that. One alternative is to use `<FloatingOverlay onClick={...}>` instead of `outsidePress` in `useDismiss`, so it naturally uses each layer that sits on top to check for "bubbling"

**Reply by stormwarning** [reply]
Is there a similar workaround for <kbd>Esc</kbd> closing all Floating elements instead of the "top"most?

**Reply by atomiks** [reply]
It works for `esc` as well, but yeah it does need focus inside the floating element to work.

**Reply by stormwarning** [reply]
Of course, just had to add an `onKeyDown` as well as the `onClick` 🤦 Thanks!

**Comment by jgodi** [comment]
I appreciate the reply. Unfortunately, we also need the document behind it
to be interactive too. The overlay would break that behavior I would think.
